stdlib/atomic.mli

```diff
@@ -61,6 +61,11 @@ val exchange : 'a t -> 'a -> 'a
     otherwise. *)
 val compare_and_set : 'a t -> 'a -> 'a -> bool
 
+(** [compare_exchange r seen v] sets the new value of [r] to [v] only
+    if its current value is physically equal to [seen] -- the comparison
+    and the set occur atomically. Returns the previous value. *)
```

Comment:
is this interface right? I guess I would have expected this to return a `Result.t` or something like that, based on the API of https://doc.rust-lang.org/std/sync/atomic/struct.AtomicUsize.html#method.compare_exchange and friends.

Reply:
maybe that's something we can just do at the call site, though.